src/linear_algebra/tensor_algebra.lean

```diff
@@ -0,0 +1,372 @@
+/-
+Copyright (c) 2020 Adam Topaz. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Author: Adam Topaz.
+-/
+
+import ring_theory.algebra
+import linear_algebra
+
+/-!
+# Tensor Algebras
+
+Given a commutative semiring R, and an R-module M, we construct the tensor algebra of M.
+This is the free R-algebra generated by the module M.
+
+## Notation
+
+1. `tensor_algebra R M` is the tensor algebra itself.
+  It is endowed with an R-algebra structure.
+2. `univ R M` is the R-linear map `M → tensor_algebra R M`.
```

Comment:
This almost feels like `has_coe M (tensor_algebra R M)` - is that a reasonable use of `coe`?

Reply:
It's important for this map to be linear, so a plain coercion is probably not enough.
But I agree with the comment about the name. This is meant to be `univ` as in `universal` -- in this case it's the universal map arising from some application of Yoneda. Another name is the "classifying map", but I guess `class` is even worse than `univ`. Do you have suggestions for another name?

Reply:
This goes badly against my principle of hoping to have easily readable names, but often the the category theory library the universal maps are called just `ι` or `π`. (Depending roughly on whether they are into or out of the universe object.)

So this could be `tensor_algebra.ι R M`.

Reply:
Done! Changed to \iota.